### ISSUE TITLE
"gettype" is not working for validating "types compatibility"

### ISSUE THREAD

**Issue by voku** [kalessil/phpinspectionsea#1166]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug                          |
| Plugin         | Php Inspections (EA Ultimate), v2.0.11 |
| Language level | PHP 7.1                        |

Code
--
```php
    /**
     * @param mixed  $var
     *
     * @return mixed
     */
    public static function filter($var)
    {
        switch (\gettype($var)) {
            case 'array':
                foreach ($var as $k => &$v) {
                    $v = self::filter($v);
                }
                unset($v);

                break;
            case 'object':
                foreach ($var as $k => &$v) {
                    $v = self::filter($v);
                }
                unset($v);

                break;
            case 'string':
                $var = foo($var);

                break;
        }

        return $var;
    }
```

Current behaviour (description/screenshot:)
--
- Expressions' type contains 'mixed', please specify possible types instead (best practices).

Expected behaviour
--
- not error

Environment details
---
PhpStorm 2018.3.3
Build #PS-183.5153.36, built on January 9, 2019
Licensed to Portable UTF-8 /  
Subscription is active until April 5, 2019
For non-commercial open source development only.
JRE: 1.8.0_152-release-1343-b26 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-141-generic

Notes
---
- Bugs fixed with the highest priority, so features might be postponed. But it shouldn't stop you, right?
- The project support is powered by community funding via [Patreon](https://www.patreon.com/kalessil)

**Comment by kalessil** [comment]
Checked, there is no way on our end to solve this. But the inspection has setting, one of them disables complaining about  mixed types.
